### PR TITLE
Fix errors when alter materialized view which based on dup table

### LINKED ISSUE #4374: Failed to drop column on materialized view
**Describe the bug**
Drop column on materialized view by AlterClause.
The alter job is failed with error msg: Create replicas failed.

**To Reproduce**
1. create table 

```
CREATE TABLE `test_sys_materialized_view_2_test_drop_column_with_mv_tb` (
  `k0` boolean NOT NULL COMMENT "",
  `k1` tinyint(4) NOT NULL COMMENT "",
  `k2` smallint(6) NOT NULL COMMENT "",
  `k3` int(11) NOT NULL COMMENT "",
  `k4` bigint(20) NOT NULL COMMENT "",
  `k5` largeint(40) NOT NULL COMMENT "",
  `k6` decimal(9, 3) NOT NULL COMMENT "",
  `k7` char(5) NOT NULL COMMENT "",
  `k8` date NOT NULL COMMENT "",
  `k9` datetime NOT NULL COMMENT "",
  `k10` varchar(20) NOT NULL COMMENT "",
  `k11` double NOT NULL COMMENT "",
  `k_new_1` int(11) NULL COMMENT "",
  `k_new` float NULL COMMENT "",
  `k12` float NOT NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`k0`, `k1`, `k2`)
COMMENT "OLAP"
PARTITION BY RANGE(`k1`)
(PARTITION p1 VALUES [("-128"), ("-10")),
PARTITION p2 VALUES [("-10"), ("0")),
PARTITION p3 VALUES [("0"), ("10")),
PARTITION p4 VALUES [("10"), ("20")))
DISTRIBUTED BY HASH(`k4`, `k5`) BUCKETS 10
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
```

2. create materialized view 

```
CREATE MATERIALIZED VIEW test_sys_materialized_view_2_test_drop_column_with_mv_index AS SELECT k0, max(k9), min(k11), sum(k12), hll_union(hll_hash(k7)) from test_sys_materialized_view_2_test_drop_column_with_mv_tb group by k0
```

3. drop column k12 on mv

```
ALTER TABLE test_sys_materialized_view_2_test_drop_column_with_mv_tb DROP COLUMN k12 FROM test_sys_materialized_view_2_test_drop_column_with_mv_index
```

4. alter job is failed

```
Create replicas failed. Error: Error replicas:xxxx
```

**Expected behavior**
Alter job is successful.

### PR DESCRIPTION
**Pull request by EmmyMiao87**
## Proposed changes

1. Input the correct keys type when mv is updated.
The keys type of mv should be used in schema change job rather then keys type of base table.
Otherwise, the be will core and thrown exception "Create replicas failed".

2. Forbidden add non-key column on agg mv directly when base table is duplicate model
If a dup table has a agg mv, user will not add a non-key column on mv.
The non-key column can only be added to dup index.

Fixed #4374
Change-Id: I375f695b3109e513622110cf0ca04a4034780909

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -556,6 +556,10 @@ private void addColumnInternal(OlapTable olapTable, Column newColumn, ColumnPosi
                 throw new DdlException("Can not assign aggregation method on column in Duplicate data model table: " + newColName);
             }
             if (!newColumn.isKey()) {
```
Comment (wutiangan) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:558]: why not put "from lint 559 to line 562" outside of if stmt?

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java [hunk 2]
```diff
@@ -237,7 +238,9 @@ protected void runPendingJob() throws AlterCancelException {
                     short shadowShortKeyColumnCount = indexShortKeyMap.get(shadowIdxId);
                     List<Column> shadowSchema = indexSchemaMap.get(shadowIdxId);
                     int shadowSchemaHash = indexSchemaVersionAndHashMap.get(shadowIdxId).schemaHash;
-                    int originSchemaHash = tbl.getSchemaHashByIndexId(indexIdMap.get(shadowIdxId));
+                    long originIndexId = indexIdMap.get(shadowIdxId);
+                    int originSchemaHash = tbl.getSchemaHashByIndexId(originIndexId);
```
Comment (wutiangan) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java:242]: can this variable be remove?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 3]
```diff
@@ -556,7 +556,9 @@ public KeysType getKeysType() {
     }
 
     public KeysType getKeysTypeByIndexId(long indexId) {
-        return indexIdToMeta.get(indexId).getKeysType();
+        MaterializedIndexMeta indexMeta = indexIdToMeta.get(indexId);
+        Preconditions.checkNotNull(indexMeta);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:560]: You can add some message if you want to check it.

```suggestion
        Preconditions.checkNotNull(indexMeta, "index id: " + indexId);
```

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java (hunk 1 above)
Comment (EmmyMiao87) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:558]: I didn't understand what you mean...

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**